internal/wildcard/match.go

```diff
@@ -81,3 +81,59 @@ func deepMatchRune(str, pattern []rune, simple bool) bool {
 	}
 	return len(str) == 0 && len(pattern) == 0
 }
+
+// Match -  finds whether the text matches/satisfies the pattern string.
+// supports  '*' and '?' wildcards in the pattern string.
+// unlike path.Match(), considers a path as a flat name space while matching the pattern.
+// The difference is illustrated in the example here https://play.golang.org/p/Ega9qgD4Qz .
+func Index(pattern, name string) (index int) {
+	if pattern == "" || pattern == "*" {
+		return 0
+	}
+	rname := make([]rune, 0, len(name))
+	rpattern := make([]rune, 0, len(pattern))
+	for _, r := range name {
+		rname = append(rname, r)
+	}
+	for _, r := range pattern {
+		rpattern = append(rpattern, r)
+	}
+	simple := false // Does extended wildcard '*' and '?' match.
+	return deepIndexRune(rname, rpattern, simple, 0)
+}
+
+func deepIndexRune(str, pattern []rune, simple bool, counter int) int {
```

Comment:
Add a short comment on what this does... Looking for pattern inside a string? why is deep needed/what does it mean?  What kind of index is returned, what kind of pattern is used?

Reply:
fixed